test/integration/cds_integration_test.cc

```diff
@@ -222,6 +222,117 @@ TEST_P(CdsIntegrationTest, CdsClusterTeardownWhileConnecting) {
   EXPECT_LE(cx_counter->value(), 1);
 }
 
+class LazyInitClusterStatsTest : public CdsIntegrationTest {
+
+public:
+  void initialize() {
+    use_real_stats_ = true;
+    config_helper_.addConfigModifier([](::envoy::config::bootstrap::v3::Bootstrap& bootstrap) {
+      bootstrap.set_enable_lazyinit_stats(true);
+    });
+    CdsIntegrationTest::initialize();
+    test_server_->waitForCounterGe("cluster_manager.cluster_added", 1);
+  }
+
+  void sendRequestToCluster1AndWaitForResponse() {
+    BufferingStreamDecoderPtr response = IntegrationUtil::makeSingleRequest(
+        lookupPort("http"), "GET", "/cluster1", "", downstream_protocol_, version_, "foo.com");
+    ASSERT_TRUE(response->complete());
+    cleanupUpstreamAndDownstream();
+  };
+
+  void updateCluster1() {
+    envoy::config::cluster::v3::Cluster cluster1_updated = cluster_creator_(
+        ClusterName1, fake_upstreams_[UpstreamIndex2]->localAddress()->ip()->port(),
+        Network::Test::getLoopbackAddressString(ipVersion()), "ROUND_ROBIN");
+
+    sendDiscoveryResponse<envoy::config::cluster::v3::Cluster>(
+        Config::TypeUrl::get().Cluster, {cluster1_updated}, {cluster1_updated}, {}, "42");
+  }
+};
+
+INSTANTIATE_TEST_SUITE_P(IpVersionsClientTypeDelta, LazyInitClusterStatsTest,
+                         DELTA_SOTW_GRPC_CLIENT_INTEGRATION_PARAMS);
+
+// Test that LazyInitTrafficStats gets created and updated correctly.
+TEST_P(LazyInitClusterStatsTest, LazyInitTrafficStatsWithClusterCreateUpdateDelete) {
+  initialize();
+
+  EXPECT_EQ(test_server_->gauge("cluster.cluster_1.ClusterTrafficStats.initialized")->value(), 0);
+  EXPECT_EQ(test_server_->counter("cluster.cluster_1.upstream_cx_total"), nullptr);
+
+  sendRequestToCluster1AndWaitForResponse();
+  // Cluster_1 trafficStats updated.
+  EXPECT_EQ(test_server_->gauge("cluster.cluster_1.ClusterTrafficStats.initialized")->value(), 1);
```

Comment:
How do we feel about this stat, which is internal to the stats infrastructure, being exposed to users in the /stats endpoint and in sinks such as Prometheus or statsd? Could it be confusing?

One thing we could do is grab a bit from the stats Flags mask and suppress this in stats sinks and admin /stats. I think we mentioned this in a chat or f2f in the distant past.

It's a testament to how readable these tests are now that I can notice this :)

OTOH I don't think this should block this PR because lazy stats are off by default. But it would be worth a TODO, mabye in the lazy_init ctor where we are creating this.

It's a little weird also to see the name of the C++ class exposed this way.